src/pyhf/cli/rootio.py

```diff
@@ -69,14 +79,15 @@ def json2xml(workspace, output_dir, specroot, dataroot, resultprefix, patch):
         )
     from .. import writexml
 
+    _ensure_dirs(output_dir)
     os.makedirs(output_dir, exist_ok=True)
     with click.open_file(workspace, 'r') as specstream:
         spec = json.load(specstream)
         for pfile in patch:
             patch = json.loads(click.open_file(pfile, 'r').read())
             spec = jsonpatch.JsonPatch(patch).apply(spec)
-        os.makedirs(os.path.join(output_dir, specroot), exist_ok=True)
-        os.makedirs(os.path.join(output_dir, dataroot), exist_ok=True)
+        _ensure_dirs(os.path.join(output_dir, specroot))
+        _ensure_dirs(os.path.join(output_dir, dataroot))
```

Comment:
@lukasheinrich This shouldn't have been added as this was removed during the Python 3 only migration.